Party management

TheNiister opened the pull request:
Added party manager
readded quest menu
added some sprites and animations
removed levels as I broke the colliders attempting to update there sprites.
fixed some bugs with the ui.
need to fix a sew more bugs with ui
added a simple cam controller

TheNiister committed:
Party Management

Added party manager
readded quest menu
added some sprites and animations
removed levels as I broke the colliders attempting to update there sprites.
fixed some bugs with the ui.
need to fix a sew more bugs with ui

TheNiister committed:
Cam Controller

TheNiister committed:
My last commit was broken

fixed my commit and added the cam controller

Shiinoya commented on Assets/Scripts/CamController.cs:
```diff
@@ -0,0 +1,23 @@
+            transform.position += new Vector3(camSpeed * Time.deltaTime * Input.GetAxis("Horizontal"), camSpeed * Time.deltaTime * Input.GetAxis("Vertical"));
+        }
+
+        if(Input.GetAxis("Mouse ScrollWheel") != 0)
+        {
+            Camera.main.orthographicSize -= Input.GetAxis("Mouse ScrollWheel") * camZoom * Time.deltaTime;
+        }
+    }
```
I wasn't able to get the camera to zoom in/out with the mouse :\

TheNiister replied:
hmm not sure why. scroll wheel works for me so ill have to look at what's going on with that

Shiinoya commented on Assets/Scripts/Questing/Party/NPC/AttackScripts/Scripts/NPC_ArcherAttack.cs:
```diff
@@ -71,5 +71,6 @@ public override void EndAttack()
     public void AttackTarget()
     {
         aTarget.currentTarget.GetComponent<HitPointManager>().EffectHealth(attackDamge);
+        thisAdventurer.GetComponent<Animator>().SetTrigger("Attack");
     }
```
I guess this also applies to treasure? Because the archer can pick up the treasure from afar.

TheNiister replied:
yeah, they just attack the loot

Shiinoya reviewed (changes requested):
Nice job! Really like how the drag and drop works.

Just a few bugs I saw. Also, some of the level walls don't have collisions? Because the units can phase through them.

TheNiister commented:
Yeah I'm aware of the collision issue. I fixed it in the second version I requested a pull for. I wasn't attempting to upload the new levels in this version but they were put in because of the way LDtk works. I've altered the levels and fixed the collisions in the latest pull

Shiinoya reviewed (approved):
Okay, let's just get this merged and pick up the pieces later 😝
